### PR TITLE
answer expression full, reference type, and choice column basic support 

### LINKED ISSUE #1039: Support for answerExpression extension
**Is your feature request related to a problem? Please describe.**
Issue: #979 

When building a data collection questionnaire, there is often a need to filter the options in a drop down list based on the selected answer to an earlier list. For example: Country > State > City - see #979 

**Describe the solution you'd like**
Enable dynamic filtering of answerOptions using answerExpression with FHIRPath - see [here](http://build.fhir.org/ig/HL7/sdc/expressions.html)

In the scenario in issue #979, where there are Country > State > City, each of which the answerOptions comes from a valueSet, the following should happen:

- If country is USA
- Populate the State lists with ValueSet of states from USA
- When changing the value of the State, the City list is updated

**Describe alternatives you've considered**
Using nested set of enableWhen components

**Other information**
For Ux example of dynamically updating lists see issue #979

### PR DESCRIPTION
**Pull request by maimoonak**
Fixes #1039 

**Description**
Implement answer expression support and example in datacapture, refereence field type and basic implementation of choice column

**Type**
Choose one: Feature

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by Tarun-Bhardwaj** [comment]
@aditya-07 , could you please review this PR?

**Review by jingtang10 — CHANGES_REQUESTED** [review]

File: datacapture/build.gradle.kts [hunk 1]
```diff
@@ -80,6 +106,7 @@ dependencies {
   implementation(Dependencies.objectDetection)
   implementation(Dependencies.objectDetectionCustom)
   implementation(Dependencies.timber)
+  implementation(project(":engine"))
```
[the remark this answers is not in the document]
Comment (Tarun-Bhardwaj) [reply datacapture/build.gradle.kts:106]: @maimoonak , have the comments above from @jingtang10 been addressed?

[the remark this answers is not in the document]
Comment (maimoonak) [reply datacapture/build.gradle.kts:106]: Done. 
PR engine = https://github.com/google/android-fhir/pull/1479
PR capture = https://github.com/google/android-fhir/pull/1480

[the remark this answers is not in the document]
Comment (maimoonak) [reply datacapture/build.gradle.kts:106]: @Tarun-Bhardwaj @jingtang10 @f-odhiambo @dubdabasoduba

**Comment by Tarun-Bhardwaj** [comment]
@maimoonak , is there an ETA by when we can merge this PR?

**Comment by aditya-07** [comment]
@maimoonak Should this PR be closed since its now split into two separate ones?

**Comment by maimoonak** [comment]
closing in favor of #1480

**Pull request closed without merging**